samples/tfm/tfm_secure_peripheral/CMakeLists.txt

```diff
@@ -8,7 +8,45 @@ cmake_minimum_required(VERSION 3.20)
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 
+if(NOT CONFIG_BUILD_WITH_TFM)
+  message(WARNING "Unsupported configuration!! CONFIG_BUILD_WITH_TFM is not enabled in this build"
+                  " which is required. Either the secure only variant of the board is selected or"
+                  " the CONFIG_BUILD_WITH_TFM was disabled through menuconfig. Make sure that you"
+                  " build with the non-secure variant of your board (target ends with /ns) and"
+                  " that you have CONFIG_BUILD_WITH_TFM enabled in your build. It is advised to"
+                  " build with pristine build (append -p in the build) after you encounter this .")
```

Comment:
`append -p in the build`: this makes the assumption that users are building using `west`, but they could be using VSCode, CMake + ninja, or even twister.

Updated text is better than previous iteration :+1: